src/rod/__init__.py

```diff
@@ -51,12 +51,42 @@ def _is_editable():
     return rod_package_dir not in site.getsitepackages()
 
 
-# Initialize the logging verbosity depending on the installation mode.
-logging.configure(
-    level=logging.LoggingLevel.DEBUG if _is_editable() else logging.LoggingLevel.WARNING
-)
+def get_default_logging_level(env_var: str) -> logging.LoggingLevel:
+    """
+    Get the default logging level.
+
+    Args:
+        env_var: The environment variable to check.
+
+    Returns:
+        The logging level to set.
+    """
+
+    import os
+
+    try:
+        return logging.LoggingLevel[os.environ[env_var].upper()]
+
+    # Raise if the environment variable is set but the logging level is invalid.
+    except AttributeError:
+        msg = f"Invalid logging level defined in {env_var}: '{os.environ[env_var]}'"
+        raise ValueError(msg)
+
+    # If the environment variable is not set, return the logging level depending
+    # on the installation mode.
+    except KeyError:
+        return (
+            logging.LoggingLevel.DEBUG
+            if installation_is_editable()
+            else logging.LoggingLevel.WARNING
+        )
+
+
+# Configure the logger with the default logging level.
+logging.configure(level=get_default_logging_level(env_var="ROD_LOGGING_LEVEL"))
```

Comment:
If the environment variable can only be `ROD_LOGGING_LEVEL`, is there a reason why you're passing it as the argument of this function? Can it just be hardcoded?

```suggestion
def get_default_logging_level() -> logging.LoggingLevel:
    """
    Get the default logging level.

    Returns:
        The logging level to set.
    """

    import os

    try:
        return logging.LoggingLevel[os.environ["ROD_LOGGING_LEVEL"].upper()]

    # Raise if the environment variable is set but the logging level is invalid.
    except AttributeError:
        msg = f"Invalid logging level defined in 'ROD_LOGGING_LEVEL'"
        raise ValueError(msg)

    # If the environment variable is not set, return the logging level depending
    # on the installation mode.
    except KeyError:
        return (
            logging.LoggingLevel.DEBUG
            if installation_is_editable()
            else logging.LoggingLevel.WARNING
        )


# Configure the logger with the default logging level.
logging.configure(level=get_default_logging_level()
```

Reply:
Having the environment variable explicitly defined makes it more visible to users rather than hiding it inside the logic of a function. Since we do not have documentation, I valued readability.

Reply:
It makes sense, thanks for the explanation!